documentation/src/docs/controller-service.md

```diff
@@ -795,73 +750,67 @@ implements the business logic for processing each request.
-Commit transaction workflow is implemented on commit event processor and
-thereby processed asynchronously. The commit transaction workflow checks for eligibility of transaction to be committed, and if true,
-it performs the commit workflow with indefinite retries until it
-succeeds. If the Transaction is not eligible for commit, which typically
-happens if Transaction is created on a new epoch while the old epoch is
-still active, then such events are reposted into the internal stream to
-be picked later.
-
-Once a Transaction is committed successfully, the record for the
-transaction is removed from under its epoch root. Then if there is an
-ongoing scale, then it calls to attempt to complete the ongoing scale.
-Trying to complete scale hinges on ability to delete old epoch which can
-be deleted if and only if there are no outstanding active transactions against the said epoch (refer to scale workflow for more details).   
+Upon receiving request to commit a Transaction, Controller Service passes the request to Transaction Utility module. This module first tries to mark the Transaction for commit in the Transaction specific metadata record via metadata store.
+
+Following this, it posts a commit Event in the internal Commit Stream. The commit event only captures the epoch in which the Transaction has to be committed. Commit Transaction workflow is implemented on commit Event processor and thereby processed asynchronously.
+
+- When commit workflow starts, it opportunistically collects all available Transactions that have been marked for commit in the given epoch and proceeds to commit them in order and one Transaction at a time.
+- A Transaction commit entails merging the Transaction Segment into its parent Segment. This works perfectly in absence of scale.
+     - However, because of scaling of a Stream, some of the parent Segments for Transaction's shadow Stream Segments could have been sealed away.
+     - In such instance, when we attempt to commit a Transactions we may not have parent Segments in which Transaction Segments could be merged into.
+     - One approach to mitigate this could have been to prevent scaling operation while there were ongoing Transactions. However, this could stall scaling for arbitrarily large periods of time and would be detrimental. Instead, controller decouples scale and Transactions and allows either to occur concurrently without impacting workings of the other.This is achieved by using a scheme called **Rolling Transactions**.
+
+#### Rolling Transactions
+
+This is achieved by using a scheme (Rolling Transactions) where controller allows Transaction Segments to outlive their parent Segments and whenever their commits are issued, at a logical level controller elevates the Transaction Segments as first class Segments and includes them in a new epoch in the epoch time series of the Stream.
+
+1. Transactions are created in an older epoch and when they are attempted to be committed.
```

Comment:
The breakdown of points here is wrong. Point 1 ends with:

```
and when they are attempted to be committed.
```

And point number 2 starts with:

```
The latest epoch is sealed, Transactions are rolled over 
```

Please fix this as the sentences are broken.

Reply:
Accepted suggestion